compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/FirMetadataSource.kt

```diff
@@ -10,55 +10,24 @@ import org.jetbrains.kotlin.fir.declarations.*
 import org.jetbrains.kotlin.ir.declarations.MetadataSource
 import org.jetbrains.kotlin.name.Name
 
-interface FirMetadataSource : MetadataSource {
-
+sealed class FirMetadataSource(open val fir: FirDeclaration) : MetadataSource {
```

Comment:
`open val` here + `override val` in subclass means there will be two fields with the same value. Maybe make it abstract here (or make it final, and remove `override val` in subclasses)? Same for `DescriptorMetadataSource`

Reply:
`final` doesn't work because then the subclasses will not have more specific types. `abstract` is indeed the best option here, thanks.